servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouteConversions.java

```diff
@@ -143,15 +138,16 @@ public void flush() throws IOException {
                         };
                         try {
                             original.handle(ctx, request.toIterable(), grpcPayloadWriter);
+
+                            // The user code has returned successfully, complete the processor so the response stream
+                            // can complete. If the user handles the request asynchronously (e.g. on another thread)
+                            // they are responsible for closing the payloadWriter.
+                            exceptionProcessor.onComplete();
                         } catch (Throwable t) {
-                            concurrentTerminalSubscriber.onError(t);
-                        } finally {
                             try {
-                                grpcPayloadWriter.close();
```

Comment:
@idelpivnitskiy - note this change. we don't auto-close in `BlockingStreamingToStreamingService` so I think we should be consistent here. we can consider adding an option to auto-close for both http & grpc, but lets do this as a followup PR if necessary.